management/server/account.go

```diff
@@ -307,6 +308,53 @@ func (a *Account) GetGroup(groupID string) *Group {
 	return a.Groups[groupID]
 }
 
+// GetExpiredPeers returns peers tha have been expired
+func (a *Account) GetExpiredPeers() []*Peer {
+	var peers []*Peer
+	for _, peer := range a.GetPeersWithExpiration() {
+		expired, _ := peer.LoginExpired(a.Settings.PeerLoginExpiration)
+		if expired {
+			peers = append(peers, peer)
+		}
+	}
+
+	return peers
+}
+
+// GetNextPeerExpiration returns the minimum duration in which the next peer of the account will expire and true if it was found.
+// If there is no peer that expires this function returns false and a zero time.Duration
+// This function only considers peers that haven't been expired yet and connected.
+func (a *Account) GetNextPeerExpiration() (time.Duration, bool) {
+	nextExpiry := time.Duration(1<<63 - 1) // max duration
+	peersWithExpiry := a.GetPeersWithExpiration()
+	if len(peersWithExpiry) == 0 {
+		return nextExpiry, false
+	}
+	for _, peer := range peersWithExpiry {
+		// consider only connected peers because others will require login on connecting to the management server
+		if peer.Status.LoginExpired || !peer.Status.Connected {
+			continue
+		}
+		_, duration := peer.LoginExpired(a.Settings.PeerLoginExpiration)
+		if duration < nextExpiry {
+			nextExpiry = duration
+		}
+	}
+
+	return nextExpiry, nextExpiry != time.Duration(1<<63-1)
```

Comment:
Maybe go with nil instead of max value for `nextExpiry` if no more expiring peers are available as it is more clean from a coding point of view to not return a value if there is no value available.